server/handlers/componetgenratorhelper.go

```diff
@@ -0,0 +1,647 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/gofrs/uuid"
+	"github.com/layer5io/meshery/server/helpers/utils"
+	"github.com/layer5io/meshery/server/models"
+	"github.com/layer5io/meshkit/models/events"
+	"github.com/layer5io/meshkit/models/meshmodel/entity"
+	"github.com/layer5io/meshkit/models/registration"
+	meshkitutils "github.com/layer5io/meshkit/utils"
+	"github.com/meshery/schemas/models/v1alpha3/relationship"
+	"github.com/meshery/schemas/models/v1beta1/component"
+	"github.com/meshery/schemas/models/v1beta1/model"
+	"gopkg.in/yaml.v2"
+)
+
+func (h *Handler) handleError(rw http.ResponseWriter, err error, logMsg string) {
+	h.log.Error(err)
+	http.Error(rw, logMsg, http.StatusInternalServerError)
+}
+
+func (h *Handler) sendSuccessResponse(rw http.ResponseWriter, userID uuid.UUID, provider models.Provider, message string, errMsg string, response *models.RegistryAPIResponse) {
+	if errMsg != "" {
+		if message != "" {
+			response.ErrMsg = message + ", " + errMsg
+			h.log.Info(response.ErrMsg)
+		} else {
+			h.log.Info(errMsg)
+			response.ErrMsg = "Import " + errMsg + "."
+		}
+	} else {
+		response.ErrMsg = message
+		h.log.Info(response.ErrMsg)
+	}
+	h.sendFileEvent(userID, provider, response)
+	rw.WriteHeader(http.StatusOK)
+	if err := json.NewEncoder(rw).Encode(response); err != nil {
+		h.log.Error(err)
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func createTempFile(dirPath string) (*os.File, error) {
+	tempFile, err := os.CreateTemp("", "upload-*.tar.gz")
+	if err != nil {
+		return nil, ErrCreateFile(err, "/tmp/upload-*.tar.gz")
+	}
+
+	if _, err = tempFile.Write([]byte(dirPath)); err != nil {
+		return nil, meshkitutils.ErrWriteFile(err, tempFile.Name())
+	}
+	return tempFile, nil
+}
+
+func processUploadedFile(filePath string, tempDir string, h *Handler, response *models.RegistryAPIResponse, provider models.Provider, checkOCI bool) error {
```

Comment:
Why is this being re-defined?
Try to use the `meshkit` functionality. There's already a logic for importing models when packaged as a directory.